### DISCUSSION TITLE
Copy action from another repo during workflow

### DISCUSSION THREAD

**Discussion by eb250130** [community/community#25607] [Actions]
<p>I have created an action that I want to use in multiple repositories.<br>
The action cannot be made public.<br>
I know that at the moment it’s not possible to run an action from a private/internal repository (but that it’s in the roadmap).<br>
I want to avoid having to put the action’s code in all my repositories and was wondering if there is a way to copy the action’s code into the executing repository while executing the workflow and then run it, but to not commit any changes to the repository<br>
So my workflow will have something like these steps:<br>
Workflow on R1:<br>
checkout (R1 codes)<br>
checkout R2 (the action repository)<br>
copy R2 action files to R1 folder (all local in runner, nothing committed)<br>
run action A1 from local repository</p>
<p>Is it possible?<br>
thanks.</p>

**Comment by laughedelic** [comment, marked as answer]
<p>Yes, you can use the usual <code>checkout</code> action to do it. Here is an example:</p>
<pre><code class="lang-yaml">    steps:
      - uses: actions/checkout@v2

      - name: Checkout private actions
        uses: actions/checkout@v2
        with:
          repository: org/shared-private-actions
          path: .github/shared-actions
          token: ${{ secrets.PRIVATE_ACCESS_TOKEN }}

      - uses: ./.github/shared-actions/action-name
        with:
          # inputs
</code></pre>
<p>Notice that you need to generate a token that has access to the private repository with actions and add it as an org/account secret.</p>
<p>You can have multiple actions in a single repository in separate folders, then you can check it out once and use any of them. Or you can have a repository per action, then you will need to use the checkout step several times.</p>

**Comment by eb250130** [comment]
<p>Thank you so much. It worked!</p>
